src/peft/tuners/lora/model.py

```diff
@@ -447,6 +447,15 @@ def _check_merge_allowed(self):
 
         Currently gptq quantization and replicated layers do not support merging.
         """
+        tied_target_modules = self._get_tied_target_modules(self.model)
```

Comment:
I think it's better to move this check to `BaseTuner` here:

https://github.com/huggingface/peft/blob/f3c7c6e5c1e1b543e1f006ef38b3fdab3a4fefaf/src/peft/tuners/tuners_utils.py#L359-L364

That way, all models that inherit from it, not just LoRA, benefit from the warning. For LoRA, we need of course to call `super()._check_merge_allowed()` in that case.